packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -72,12 +74,15 @@ export const useTreeItem2 = <TPlugins extends DefaultTreeViewPlugins = DefaultTr
   const createContentHandleClick =
     (otherHandlers: EventHandlers) => (event: React.MouseEvent & MuiCancellableEvent) => {
       otherHandlers.onClick?.(event);
-      if (event.defaultMuiPrevented) {
+      if (event.defaultMuiPrevented || checkboxRef.current?.contains(event.target as HTMLElement)) {
```

Comment:
We don't want to toggle the expansion when clicking on the checkbox

Reply:
Have you tried setting `event = defaultMuiPrevented` in the `checkbox` change handler? 🤔 
Maybe it would allow us to ditch the `||` case? 🤔

Reply:
Yes, the `onChange` event is not shared with the `onClick` one so it does not work
I could add an `onClick` on the checkbox and prevent there of course, I'll see if it works. If it allow us to skip the ref then it's probably an improvement

Reply:
Ah, damn, indeed, those are different kinds of events. 🙈 
In that case, I'm fine with both approaches. However, naturally, if ref is used only for this check, then it probably makes sense to intercept the `onClick` on the `checkbox` and avoid this problem that way. 👍

Reply:
I tried, the `defaultMuiPrevented` works.
The only problem is that it's usually a property our users set to disable our behavior and that we very rarely set ourselves (I only found one occurence in the core codebase).
I don't know if this would have side effects, to be honest I didn't find any...


Here is the diff for `TreeItem`:


```diff
diff --git a/packages/x-tree-view/src/TreeItem/TreeItemContent.tsx b/packages/x-tree-view/src/TreeItem/TreeItemContent.tsx
index 161d3d3af..c16d94c5b 100644
--- a/packages/x-tree-view/src/TreeItem/TreeItemContent.tsx
+++ b/packages/x-tree-view/src/TreeItem/TreeItemContent.tsx
@@ -3,6 +3,7 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import Checkbox from '@mui/material/Checkbox';
 import { useTreeItemState } from './useTreeItemState';
+import { MuiCancellableEvent } from '../internals/models/MuiCancellableEvent';
 
 export interface TreeItemContentProps extends React.HTMLAttributes<HTMLElement> {
   className?: string;
@@ -85,7 +86,6 @@ const TreeItemContent = React.forwardRef(function TreeItemContent(
   } = useTreeItemState(itemId);
 
   const icon = iconProp || expansionIcon || displayIcon;
-  const checkboxRef = React.useRef<HTMLButtonElement>(null);
 
   const handleMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
     preventSelection(event);
@@ -95,8 +95,8 @@ const TreeItemContent = React.forwardRef(function TreeItemContent(
     }
   };
 
-  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
-    if (checkboxRef.current?.contains(event.target as HTMLElement)) {
+  const handleClick = (event: React.MouseEvent<HTMLDivElement> & MuiCancellableEvent) => {
+    if (event.defaultMuiPrevented) {
       return;
     }
 
@@ -111,6 +111,12 @@ const TreeItemContent = React.forwardRef(function TreeItemContent(
     }
   };
 
+  const handleCheckboxClick = (
+    event: React.MouseEvent<HTMLButtonElement> & MuiCancellableEvent,
+  ) => {
+    event.defaultMuiPrevented = true;
+  };
+
   return (
     /* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions -- Key event is handled by the TreeView */
     <div
@@ -131,8 +137,8 @@ const TreeItemContent = React.forwardRef(function TreeItemContent(
           className={classes.checkbox}
           checked={selected}
           onChange={handleCheckboxSelection}
+          onClick={handleCheckboxClick}
           disabled={disabled || disableSelection}
-          ref={checkboxRef}
           tabIndex={-1}
         />
       )}
```